apps/admin-ui/src/component/filters/ReservationUnitFilter.tsx

```diff
@@ -1,59 +1,28 @@
-import React from "react";
-import { useTranslation } from "react-i18next";
 import { useReservationUnitsFilterParamsQuery } from "@gql/gql-types";
-import { SortedSelect } from "@/component/SortedSelect";
 import { filterNonNullable } from "common/src/helpers";
 
-type OptionType = {
-  label: string;
-  value: number;
-};
-
-type Props = {
-  onChange: (reservationUnits: OptionType[]) => void;
-  value: OptionType[];
-};
-
-// TODO this should be refactored to use Apollo cache because local state is bad
-// local state problems:
-// - two components have separate state so a mutation requires refetch on both
-// - load time issues if the data changes between component loads they are inconsistant
-// - the fetch (that could include 100s of gql queries) is run for every component
-// i.e. create dummy data of 10k ReservationUnits, add 100 filter components to the page and
-// watch the backend break.
-const ReservationUnitFilter = ({ onChange, value }: Props): JSX.Element => {
-  const { t } = useTranslation();
-
-  // TODO this request is rerun whenever the selection changes (it'll return 0 every time)
-  const { data, loading } = useReservationUnitsFilterParamsQuery({
-    // breaks the cache
-    fetchPolicy: "no-cache",
+export function useReservationUnitOptions() {
+  const { data, loading, fetchMore } = useReservationUnitsFilterParamsQuery({
+    // auto fetch more (there is no limit, expect number of them would be a few hundred, but in theory this might cause problems)
+    onCompleted: async (d) => {
+      if (d?.reservationUnits?.pageInfo.hasNextPage) {
+        await fetchMore({
+          variables: {
+            after: d.reservationUnits.pageInfo.endCursor,
+          },
+        });
+      }
+    },
   });
```

Comment:
FIX: This doesn't work for more than 200 objects. `onComplete` is only ever called once. Have to use `useEffect` for this.